src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -20,10 +20,18 @@ import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.exception.AsyncEve
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AppointmentType
 import java.net.URI
 
-interface NotifyService {
+open class NotifyService {
   fun generateResourceUrl(baseURL: String, path: String, vararg args: Any): URI {
     return UriComponentsBuilder.fromHttpUrl(baseURL).path(path).buildAndExpand(*args).toUri()
   }
+
+  fun responsibleOfficerFirstName(responsibleOfficer: ResponsibleOfficer): String {
+    return responsibleOfficer.name?.split(' ')?.first() ?: ""
+  }
+
+  fun responsibleOfficerEmail(responsibleOfficer: ResponsibleOfficer): String {
+    return responsibleOfficer.email ?: throw RuntimeException("no email address for service user's responsible officer")
```

Comment:
Could we throw an exception class that is semantically explicit about "our assumptions about received data were invalid"? For simpler monitoring. Maybe:
```suggestion
    return responsibleOfficer.email ?: throw ExternalDataExpectationFailed("no email address for service user's responsible officer")
```
or -- we could use the `InterventionsDataAssumptionInvalid` telemetry in the other comment

Reply:
like this idea - i'll have a play arond with it